enterprise/metrics/src/main/java/org/neo4j/metrics/source/db/CheckPointingMetrics.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.metrics.source.db;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.MetricRegistry;
+
+import java.util.TreeMap;
+
+import org.neo4j.kernel.impl.transaction.log.checkpoint.CheckPointEventObservable;
+import org.neo4j.kernel.impl.transaction.log.checkpoint.CheckPointEventObservable.Observer;
+import org.neo4j.kernel.impl.transaction.log.checkpoint.CheckPointerMonitor;
+import org.neo4j.kernel.lifecycle.LifecycleAdapter;
+import org.neo4j.metrics.output.EventReporter;
+import org.neo4j.metrics.source.Metrics;
+
+import static com.codahale.metrics.MetricRegistry.name;
+import static java.util.Collections.emptySortedMap;
+
+public class CheckPointingMetrics extends LifecycleAdapter implements Metrics
+{
+    private static final String CHECK_POINT_PREFIX = "neo4j.check_point";
+    public static final String CHECK_POINT_EVENTS = name( CHECK_POINT_PREFIX, "events" );
+    public static final String CHECK_POINT_TOTAL_TIME = name( CHECK_POINT_PREFIX, "total_time" );
+
+    private final boolean eventBasedReporting;
+    private final MetricRegistry registry;
+    private final CheckPointEventObservable checkPointEventObservable;
+    private final CheckPointerMonitor checkPointerMonitor;
+    private final Observer observer;
+
+    public CheckPointingMetrics( boolean eventBasedReporting, EventReporter reporter, MetricRegistry registry,
+            CheckPointEventObservable checkPointEventObservable, CheckPointerMonitor checkPointerMonitor )
+    {
+        this.eventBasedReporting = eventBasedReporting;
+        this.registry = registry;
+        this.checkPointEventObservable = checkPointEventObservable;
+        this.checkPointerMonitor = checkPointerMonitor;
+        this.observer = ( totalTimeMillis, numberOfEvents ) -> {
+            TreeMap<String,Gauge> gauges = new TreeMap<>();
```

Comment:
I need a `SortedMap` to pass down to the `codahale` library. If you know a better way of doing it, please share.